static/app/components/tabs/tab.tsx

```diff
@@ -37,64 +42,165 @@ function handleLinkClick(e: React.PointerEvent<HTMLAnchorElement>) {
   }
 }
 
+interface BaseTabProps {
+  children: React.ReactNode;
+  hidden: boolean;
+  isSelected: boolean;
+  orientation: Orientation;
+  overflowing: boolean;
+  tabProps: DOMAttributes<FocusableElement>;
+  /**
+   * Additional props to be merged with `tabProps`. This is used
+   * by <DraggableTab> to pass in props used for drag-and-drop functionality.
+   */
+  additionalProps?: React.HTMLAttributes<HTMLElement>;
+  borderStyle?: 'solid' | 'dashed';
+  to?: string;
+  variant?: 'vanilla' | 'draggable';
```

Comment:
Rather than making this a boolean, i've left it as a union type since I heard there are _even more_ tab variants that are coming.

Reply:
also open to naming these something else in case the draggable tabs design is planned to be used elsewhere

Reply:
A union is great! I would ask Vu about naming, he did seem to think that this new tab design would be used elsewhere. I _believe_ it is meant to be used when it is at the top of the page, whereas the current design would only be used for lower, self-contained elements.

Reply:
Spoke with Vu and decided on renaming it to "flat" (old, existing design) and "filled" (new design, used for draggable tabs)